auth/src/main/java/com/firebase/ui/auth/ui/provider/FacebookProvider.java

```diff
@@ -0,0 +1,58 @@
+package com.firebase.ui.auth.ui.provider;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.ViewModelProviders;
+import android.content.Intent;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.RestrictTo;
+import android.support.annotation.StringRes;
+
+import com.facebook.WebDialog;
+import com.facebook.login.LoginManager;
+import com.firebase.ui.auth.IdpResponse;
+import com.firebase.ui.auth.R;
+import com.firebase.ui.auth.data.remote.FacebookSignInHandler;
+import com.firebase.ui.auth.ui.HelperActivityBase;
+import com.firebase.ui.auth.util.data.ProviderUtils;
+import com.google.firebase.auth.FacebookAuthProvider;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public class FacebookProvider extends ProviderBase {
+    private final FacebookSignInHandler mHandler;
+
+    public FacebookProvider(HelperActivityBase activity) {
+        WebDialog.setWebDialogTheme(activity.getFlowParams().themeId);
+        mHandler = ViewModelProviders.of(activity).get(FacebookSignInHandler.class);
+        mHandler.init(ProviderUtils.getConfigFromIdpsOrThrow(
+                activity.getFlowParams().providerInfo, FacebookAuthProvider.PROVIDER_ID));
+    }
+
+    @Override
+    public LiveData<IdpResponse> getResponseListener() {
+        return mHandler.getOperation();
+    }
+
+    @StringRes
+    @Override
+    public int getNameRes() {
+        return R.string.fui_idp_name_facebook;
+    }
+
+    @Override
+    @LayoutRes
+    public int getButtonLayout() {
+        return R.layout.fui_idp_button_facebook;
+    }
+
+    @Override
+    public void startLogin(@NonNull HelperActivityBase activity) {
```

Comment:
1) nit, but for consistency this should probably be called `startSignIn` since we use the term 'sign in' not 'log in' pretty much everywhere.

2) I find it a bit strange that this class talks directly to `LoginManager`.  Shouldn't all of the Facebook SDK code be contained within the `FacebookSignInHandler` and this class merely acts as a pass-through?

Reply:
Yup, fixed.

Reply:
For the second part, I'm not doing that so we don't send the activity over to the view model... do you think we should?